samples/benchmarks/coremark/src/main.c

```diff
@@ -50,18 +49,10 @@ static const struct gpio_dt_spec status_led;
 static K_SEM_DEFINE(start_coremark, 0, 1);
 static atomic_t coremark_in_progress;
 
-static void flush_log(void)
-{
-	if (IS_ENABLED(CONFIG_LOG_PROCESS_THREAD)) {
-		while (log_data_pending()) {
-			k_sleep(K_MSEC(100));
-		}
-		k_sleep(K_MSEC(100));
-	} else {
-		while (LOG_PROCESS()) {
-		}
-	}
-}
+/* Enfore synchronous logging as the sample doesn't flush logs. */
+BUILD_ASSERT(IS_ENABLED(CONFIG_LOG_MODE_MINIMAL) || IS_ENABLED(CONFIG_LOG_MODE_IMMEDIATE),
```

Comment:
I wouldn't add this assert - we can have referred logging and use sample correctly - our current sample is prove of that.

Reply:
Assert allows to simplify logic, so maybe it's not a bad idea to change it like that (there is no reason to use this sample with deferred logging I think)

Reply:
yes, but I removed the logic for flushing logs so switching to the deferred mode would give you incorrect results. This sample is not the CoreMark module and it does not support the deferred logging after my change